ambassador/ambassador/ir/irhttpmappinggroup.py

```diff
@@ -323,14 +323,31 @@ def finalize(self, ir: 'IR', aconf: Config) -> List[IRCluster]:
                 else:
                     total_weight += mapping.weight
 
-            # OK, once that's done normalize all the weights.
+            # OK, once that's done normalize all the weights. Be sure to round each
+            # weight into an integer, since the Envoy spec expects that. We may not
+            # sum to 100 exactly by doing this, so we'll need to fix that up later.
             if unspecified_mappings:
                 for mapping in self.mappings:
                     if not mapping.get("weight", 0):
-                        mapping.weight = (100.0 - total_weight)/unspecified_mappings
+                        mapping.weight = round((100.0 - total_weight)/unspecified_mappings)
             elif total_weight != 100.0:
                 for mapping in self.mappings:
-                    mapping.weight *= 100.0/total_weight
+                    mapping.weight = round(mapping.weight * (100.0/total_weight))
+
+            final_total = 0
+            for mapping in self.mappings:
+                final_total += mapping.weight
+
+            # Spread out the remainder evenly over the mappings. Technically, if
+            # rounding is perfect, this is never necessary. But to protect against
```

Comment:
Now that I think about it, the very simple case of 33.3 + 33.3 + 33.3 would make this fixup actually necessary.

I'll fix the comment and test this by hand.

Reply:
Cleaned this up and verified that this case works.